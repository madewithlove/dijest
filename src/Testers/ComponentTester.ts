import FluentTester, { TestType } from './FluentTester';
import { ComponentFactory } from '../component';

export default class ComponentTester extends FluentTester {
    setComponent(value: ComponentFactory) {
        return this.setTested(value, TestType.Component);
    }
}
