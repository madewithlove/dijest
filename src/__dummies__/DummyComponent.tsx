import * as React from 'react';

const DummyComponent: React.SFC<{ text?: string }> = ({ text }) =>
    <div>
        <div className="some-div">hello</div>
        <button>
            {text || 'foobar'}
        </button>
    </div>;

export default DummyComponent;
