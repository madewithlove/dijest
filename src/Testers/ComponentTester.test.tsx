import * as React from 'react';
import ComponentTester from './ComponentTester';
import DummyComponent from '../__dummies__/DummyComponent';

new ComponentTester()
    .setDescription('can test components')
    .setComponent(<DummyComponent />)
    .run();

new ComponentTester()
    .setDescription('can access callback for advanced operations')
    .setComponent(<DummyComponent />)
    .run((wrapper, snapshot) => {
        snapshot(wrapper.find('.some-div'));
    });

new ComponentTester()
    .setDescription('can access callback for advanced operations')
    .setComponent(<DummyComponent text="foo" />)
    .run((wrapper, snapshot) => {
        snapshot(wrapper.find('.some-div'));
    });
