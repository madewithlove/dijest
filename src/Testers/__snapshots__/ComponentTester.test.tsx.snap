// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can access callback for advanced operations 1`] = `
<div
  className="some-div"
>
  hello
</div>
`;

exports[`can access callback for advanced operations 2`] = `
<div
  className="some-div"
>
  hello
</div>
`;

exports[`can test components 1`] = `
<div>
  <div
    class="some-div"
  >
    hello
  </div>
  <button>
    foobar
  </button>
</div>
`;
